### PR TITLE
Adding powershell scripts to generate and push chocolatey package

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit d6630be245de4f75cc1c95ec5cae4d4ace23836f (Job results: [227](https://jenkins.osquery.io/job/osqueryPullRequestOSX/227/)) failed one or more tests (**macOS/OS X**).

**Comment by PoppySeedPlehzr** [comment]
ok to test

**Comment by PoppySeedPlehzr** [comment]
This addresses #2393

File: tools/deployment/chocolatey/tools/chocolateyBeforeModify.ps1 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+function Remove-DenyWriteAcl {
+  [CmdletBinding(SupportsShouldProcess=$true,ConfirmImpact="Medium")]
+  param(
+    [string] $targetDir = ''
+  )
+  if($PSCmdlet.ShouldProcess($targetDir)){
```
Comment (theopolis) [inline tools/deployment/chocolatey/tools/chocolateyBeforeModify.ps1]: nit, inconsistent spacing.

File: tools/deployment/chocolatey/tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -0,0 +1,87 @@
+function Set-DenyWriteAcl {
```
Comment (theopolis) [inline tools/deployment/chocolatey/tools/chocolateyinstall.ps1]: nit, potentially significant code reuse, is it possible to combine into a library or import?

File: tools/deployment/chocolatey/tools/chocolateyinstall.ps1 [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  [CmdletBinding(SupportsShouldProcess=$true,ConfirmImpact="Medium")]
+  param(
+    [string] $targetDir = ''
+  )
+  if($PSCmdlet.ShouldProcess($targetDir)){
```
Comment (theopolis) [inline tools/deployment/chocolatey/tools/chocolateyinstall.ps1]: nit, same.

File: tools/deployment/chocolatey/tools/chocolateyinstall.ps1 [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+$packageParameters = $env:chocolateyPackageParameters
+$arguments = @{}
+
+# Now parse the packageParameters using good old regular expression
+if ($packageParameters) {
+  $match_pattern = "\/(?<option>([a-zA-Z]+)):(?<value>([`"'])?([a-zA-Z0-9- _\\:\.]+)([`"'])?)|\/(?<option>([a-zA-Z]+))"
+  $option_name = 'option'
+  $value_name = 'value'
+
+  if ($packageParameters -match $match_pattern ){
```
Comment (theopolis) [inline tools/deployment/chocolatey/tools/chocolateyinstall.ps1]: nit same.

File: tools/deployment/chocolatey/tools/chocolateyinstall.ps1 [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    $results = $packageParameters | Select-String $match_pattern -AllMatches
+    $results.matches | ForEach-Object {
+      $arguments.Add(
+        $_.Groups[$option_name].Value.Trim(),
+        $_.Groups[$value_name].Value.Trim())
+    }
+  }
+  else {
```
Comment (theopolis) [inline tools/deployment/chocolatey/tools/chocolateyinstall.ps1]: nit, inconsistent newline.

File: tools/deployment/chocolatey/tools/chocolateyuninstall.ps1 [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+if ((Get-Service $serviceName -ErrorAction SilentlyContinue)) {
+  Stop-Service $serviceName
+  Set-Service $serviceName -startuptype 'manual'
+  Get-CimInstance -ClassName Win32_Service -Filter "Name='osqueryd'" | Invoke-CimMethod -methodName Delete
+}
+
+if (Test-Path $targetFolder) {
+  Remove-Item -Force -Recurse $targetFolder
+  if(Test-Path $targetFolder) {
```
Comment (theopolis) [inline tools/deployment/chocolatey/tools/chocolateyuninstall.ps1]: nit, same.

File: tools/deployment/make_windows_package.ps1 [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+  $version = git describe --tags
+  $latestFullVersion = (git tag)[-2]
+  # If split len is greater than 1, this is a pre-release. Chocolatey is particular
+  # about the format of the version for pre-releases.
+  if ($version.split('-').length -eq 3) {
+    $version = $version.split('-')[0] + '-' + $version.split('-')[2]
+  }
+
+  # TODO: Hash this out for more robust directory awareness
```
Comment (theopolis) [inline tools/deployment/make_windows_package.ps1]: Why a TODO? What is the blocker?

Comment (PoppySeedPlehzr) [inline tools/deployment/make_windows_package.ps1]: I'll remove the TODO not, I wasn't entirely satisified with making assumptions with where the script is being run, however I added the `if` check for pathing after and never removed this.

**Comment by PoppySeedPlehzr** [comment]
I think this is ready. Did some testing locally and it's behaving as desired. @theopolis if you get a chance to review this I'd appreciate it. Thanks!

**Comment by osqueryer** [comment]
:-1: The commit a73b1054493780691549199cf3a780a3fe74a5e8 (Job results: [4002](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4002/)) failed one or more tests (**Linux**).

**Pull request merged**